src/azure-cli-core/azure/cli/core/_help.py

```diff
@@ -152,7 +329,13 @@ def __init__(self, help_ctx, delimiters, parser):
         help_param = next(p for p in self.parameters if p.name == '--help -h')
         help_param.group_name = 'Global Arguments'
 
+    # Todo: is this necessary? This has exactly the same behavior as superclass.
     def _load_from_data(self, data):
+
+        def _params_equal(data, param):
+            for name in param.name_source:
+                return data.get("name") == name.lstrip("-")
```

Comment:
If we're returning on the first iteration, why do we need the loop?

Reply:
This is because name_source is an array of the option names the parameter can take. E.g. --name and -n.

Reply:
I don't follow. Right now, you are only looking at the first index of the array.
Is this what you mean to do?

Reply:
@williexu, so param.name_source will be something like `[--foo, -f]`. The idea here is to match the parameter name in help.yaml with one of these option names. Moreover, we technically don't know for sure if the order of the list is `[--foo, -f]` or `[-f, --foo]`. 

Also this code affects how parameters are read from help.py as well. One, can now use just `foo` or `f` in  help.py, just as in help.yaml, instead of `--foo -f`

However, I can change the logic to accept only the longer option name, as this might be clearer than accepting either option names.

cc @tjprescott

Reply:
Thanks for clarifying @williexu

Reply:
After our talk, I believe you want to look through the full list of options, please make the fix.

Reply:
@williexu. I made and pushed the change after we spoke